compiler/modules/front/codeGen.bal

```diff
@@ -359,11 +362,42 @@ function codeGenStmts(CodeGenContext cx, bir:BasicBlock bb, Environment initialE
             env.assignments.push(...effect.assignments);
         }
     }                
+    check unusedLocalVariables(cx, env);
     int[] assignments = [];
     addAssignments(assignments, env.assignments, startRegister);
     return { block: curBlock, assignments };
 }
 
+function unusedLocalVariables(CodeGenContext cx, Environment env) returns CodeGenError? {
```

Comment:
There's a simpler way to do this. Pass in a parameter `Binding? limit` which is initialEnv.bindings in codeGenStmts. Then check the used field in each binding in env up to limit that is not a narrowed binding. You don't need localVarNames.

Reply:
Changed